RepoDb/RepoDb.Tests/RepoDb.IntegrationTests/TypeConversionsTest.cs

```diff
@@ -417,7 +417,8 @@ public void TestSqlConnectionCrudConvertionFromStringToDate()
                 var data = connection.Query<StringToDateClass>(e => e.SessionId == (Guid)id).FirstOrDefault();
 
                 // Assert
-                Assert.AreEqual("1/1/1970 12:00:00 AM", data.ColumnDate);
+                Assert.IsTrue(DateTime.Parse(entity.ColumnDate)
```

Comment:
@calumchisholm the reason why I am not parsing is to ensure the string equality. I have to assert the String here, not the DateTime type. Can we revert this one?

Reply:
The string returned previously was dependent upon the user's current locale, so the test failed for me. We didn't all have string equality before either :-)

The corresponding change in FunctionFactory uses [DateTime.ToString("o") format specifier](https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings#Roundtrip) to return locale-agnostic, time-zone-aware (ISO-8601) date strings, which avoids this problem.

The only way to reliably perform round-trip string comparisons for all locales would be to specify our initial dates using this same ISO-8601 format. I can create a further PR for this if you're happy to go with that approach?

Reply:
I agree about the locale issue. But I will test this one first, I need to apply the changes you made in the FunctionFactory regarding DateTime.ToString("o").